File: 171. Excel Sheet Column Number/sol.cpp
class Solution {
public:
    int titleToNumber(string s) {
        int len = s.size(), ret = 0;
        for(int i = 0; i < len; ++i){
            ret = ret * 26 + getVal(s[i]);
        }
        return ret;
    }
    int getVal(char c){
        return c - 'A' + 1;
    }
};

File: 0007.ReverseInteger/sol.java
class Solution {
    public int reverse(int x) {
        int ret = 0;
        while(x != 0){
            int pop = x % 10;
            x = x / 10;
            //max:2147483647 min:-2147483648
            if(ret > Integer.MAX_VALUE / 10 || (ret == Integer.MAX_VALUE / 10 && pop > 7)){
                return 0;
            }
            if(ret < Integer.MIN_VALUE / 10 || (ret == Integer.MIN_VALUE / 10 && pop < -8)){
                return 0;
            }
            ret = ret * 10 + pop;
        }
        return ret;
    }
}